app/HistoryExporter.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Exporters;
+
+use App\Repositories\HistoryRepository;
+
+class HistoryExporter
+{
+    protected $historyRepository;
+
+    public function __construct()
+    {
+        $this->historyRepository = new HistoryRepository();
+    }
+
+    public function export($date = null)
+    {
+        $data = $this->historyRepository->allGroupedBy('date');
+        $item = (is_null($date)) ? $data : [$date => $data[$date]];
+
+        return $this->showHistoryItems($item);
```

Comment:
let's implement method `get` inside the repository to get array of records, filtered by the criteria

```suggestion
        return (is_null($date))
            ? $this->exportByDate($date)
            : $this->exportAll();
```

```php
//exportByDate

$data = $this->historyRepository->get(['date' => $date]);

return $this->showHistoryItems([$date => $data]);
```